business_objects/project.py

```diff
@@ -60,6 +66,38 @@ def get_label_distribution(
         return values[0]
 
 
+def get_confidence_distribution(
+    project_id: str,
+    labeling_task_id: str,
+    data_slice_id: str = None,
+    num_samples: int = None,
+) -> List[float]:
+    query_filter = session.query(RecordLabelAssociation.confidence).filter(
+        RecordLabelAssociation.project_id == project_id,
+        LabelingTask.id == labeling_task_id,
+        LabelingTask.id == LabelingTaskLabel.labeling_task_id,
+        RecordLabelAssociation.labeling_task_label_id == LabelingTaskLabel.id,
+        RecordLabelAssociation.source_type == enums.LabelSource.WEAK_SUPERVISION.value,
+    )
+
+    if data_slice_id is not None:
+        query_filter = query_filter.filter(
+            DataSliceRecordAssociation.data_slice_id == data_slice_id,
+            DataSliceRecordAssociation.record_id == RecordLabelAssociation.record_id,
+            DataSliceRecordAssociation.project_id == project_id,
+        )
+
+    if num_samples is not None:
+        query_filter = query_filter.limit(num_samples)
```

Comment:
i think for a limit we should also at least order by random to not create a pseudo-deterministic state

Reply:
adding a random order would result in pulling different data whenever you reload the overview page -> can't compare to previous state anymore. picking the first N as the random sample is ok in this case.

Reply:
the first N are also not deterministic that s why I wouldn't want the pseudo determinisitc state that looks like a comparable state but isn't --> resolve if still irrelevant :)

Reply:
understood. maybe not completely irrelevant, and interesting for later, but good for now :)